Add custom sorting documentation

matthiasseghers opened the pull request:
| Q | A
| --- | ---
| Fixed tickets | N/A
| Related PRs | N/A
| License | MIT

#### What's in this PR?
Adds a section that explains how to customize the fields of smart_content that are displayed on the admin side. By making small adjustments, you can change the configuration without having to write a custom data provider with a custom entity. This can be particularly helpful when sorting needs to be done on a custom property for one of the types, or when certain functionality is fixed. For instance, you may want to overwrite the sorting method without giving the option to overwrite it with a custom QueryBuilder. These changes to the documentation file are meant to clarify and simplify the process of customizing smart_content fields on the admin side.

#### Why?

The documentation currently doesn't mention that certain parts of the DataProvider can be overwritten to make minor changes to the configuration without having to create a custom data provider. Additionally, it doesn't explain how to sort by a custom property based on one of the templates.

matthiasseghers committed:
Add custom sorting documentation

alexander-schranz commented on cookbook/custom-page-data-provider.rst:
```diff
@@ -103,3 +103,106 @@ is implemented, which filters the pages for a specific author:
     </property>
+
+Adding custom sorting
+---------------------
+You can implement custom sorting criteria by creating a new ``DataProvider`` class that extends from the ``PageDataProvider`` class and using it instead.
```
```suggestion
---------------------

You can implement custom sorting criteria by creating a new ``DataProvider`` class that extends from the ``PageDataProvider`` class and using it instead.
```

alexander-schranz committed:
Update cookbook/custom-page-data-provider.rst

matthiasseghers committed:
Update custom-page-data-provider.rst

Use the ProviderConfiguration from the parent and set/overwrite values using setters.

matthiasseghers committed:
Merge branch 'smartcontent-custom-sorting' of https://github.com/matthiasseghers/sulu-docs into smartcontent-custom-sorting

matthiasseghers committed:
Fix docs

matthiasseghers committed:
Keep code snippets on topic.

alexander-schranz commented:
@matthiasseghers thank you!
